pkg/kubectl/cmd/taint/taint.go

```diff
@@ -39,8 +39,8 @@ import (
 	"k8s.io/kubernetes/pkg/kubectl/util/i18n"
 )
 
-// TaintOptions have the data required to perform the taint operation
-type TaintOptions struct {
+// Options have the data required to perform the taint operation
+type Options struct {
```

Comment:
`TaintOptions` needs to be renamed as `Options` everywhere. Tests should pass after that :)

Reply:
Apologies to the PR author, golint conflicts with the guidance given in CLI conventions dev doc (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-cli/kubectl-conventions.md#command-implementation-conventions)

this has come up repeatedly (a couple examples - https://github.com/kubernetes/kubernetes/pull/78634#discussion_r290782303, https://github.com/kubernetes/kubernetes/pull/75202#pullrequestreview-231460270)

These exported types are used by downstream consumers, and will be used by many more once they are moved to k8s.io/kubectl. I don't think satisfying golint is a good reason to rename them.

Reply:
Note that the golint documentation specifically states `do not expect or require code to be completely "lint-free"`

From https://github.com/golang/lint#purpose:

> The suggestions made by golint are exactly that: suggestions. Golint is not perfect, and has both false positives and false negatives. Do not treat its output as a gold standard. We will not be adding pragmas or other knobs to suppress specific warnings, so do not expect or require code to be completely "lint-free". In short, this tool is not, and will never be, trustworthy enough for its suggestions to be enforced automatically, for example as part of a build process.

Reply:
thanks @liggitt I will revert it back to TaintOptions then.

Reply:
I [started a thread on the sig-cli list](https://groups.google.com/forum/#!topic/kubernetes-sig-cli/l8tmJgozneo) to try to resolve this. If you want to wait until that is decided, that might be best. Sorry for the churn.

Reply:
yes, and yet we enforce it on all of our repos where we tend to generate code known to trivially fail it :disappointed:

Reply:
👍I will wait then, thank you for starting the thread.